scripts/generate_rust_analyzer.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env python3
+import sys
+import os
+from os import path
+import json
+
+if (len(sys.argv) != 5):
+    print("Not enough arguments")
+    sys.exit(-1)
+
+srctree = sys.argv[1]
+objtree = sys.argv[2]
+lib_src = sys.argv[3]
+bindgen_file = sys.argv[4]
+
+crate_list = []
+
+def find(list, name):
+    pos = -1
+    for x in list:
+        pos += 1
+        if x['display_name'] == name:
+            return pos
+
+def generate_deps(source, deps):
+    ret = []
+    for x in deps:
+        ret.append({"crate": find(source, x), "name": x})
+    return ret
```

Comment:
I usually avoid comprehensions when things are complex, but this is pretty much the best case for them:

```suggestion
    return [{"crate": find(source, x), "name": x} for x in deps]
```

Another option is `map()` with a lambda etc.

Reply:
I use that option. Don't know how to use map, and I think this way is readable enough.